src/components/PanelV2/__tests__/PanelV2.spec.js

```diff
@@ -3,38 +3,121 @@
  * @copyright IBM Security 2019
  */
 
-import { mount } from 'enzyme';
-import React, { Fragment } from 'react';
-
-import { label, onClick } from '../_mocks_';
+import { cleanup, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
 
 import { Button, PanelV2, PanelContent } from '../../..';
 
 describe('PanelV2', () => {
-  it('renders', () => {
-    const wrapper = mount(
+  afterEach(cleanup);
+
+  test('should render as expected', async () => {
+    render(
       <PanelV2
-        title={label}
-        subtitle={label}
-        closeButton={{ onClick }}
-        renderFooter={() => (
-          <Fragment>
-            <Button
-              id="example-secondary-button"
-              kind="secondary"
-              onClick={onClick}
-            >
-              Close
-            </Button>
-            <Button id="example-primary-button" onClick={onClick}>
-              Add
-            </Button>
-          </Fragment>
-        )}
+        title="test title"
+        subtitle="test subtitle"
+        closeButton={{
+          label: 'test close',
+        }}
+        renderFooter={() => <Button>test footer button</Button>}
       >
-        <PanelContent>{label}</PanelContent>
+        <PanelContent>test content</PanelContent>
       </PanelV2>
     );
-    expect(wrapper).toMatchSnapshot();
+    await expect(document.body).toMatchSnapshot();
+  });
+
+  test('should have no Axe or DAP violations', async () => {
+    const main = document.createElement('main');
+    render(
+      <PanelV2
+        title="test title"
+        subtitle="test subtitle"
+        closeButton={{
+          label: 'test close',
+        }}
+        renderFooter={() => <Button>test footer button</Button>}
+      >
+        <PanelContent>test content</PanelContent>
+      </PanelV2>,
+      {
+        // DAP requires a landmark '<main>' in the DOM:
+        container: document.body.appendChild(main),
+      }
+    );
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('PanelV2');
+  });
+
+  test('should invoke close mock when close button is clicked', () => {
+    const onCloseMock = jest.fn();
+    const { getByLabelText } = render(
+      <PanelV2
+        closeButton={{
+          label: 'test close',
+          onClick: onCloseMock,
+        }}
+      />
+    );
+
+    userEvent.click(getByLabelText(/test close/i));
+    expect(onCloseMock).toHaveBeenCalled();
+  });
+
+  test('should not render panel when `isOpen` is set to `false`', () => {
+    const { queryByText } = render(
+      <PanelV2
+        title="test title"
+        isOpen={false}
+        closeButton={{
+          label: 'test close',
+        }}
+      />
+    );
+
+    expect(queryByText(/test title/i)).not.toBeInTheDocument();
+  });
+
+  it('should cycle panel elements in tab order', () => {
```

Comment:
Does this actually test `PanelV2` logic or is it testing some focus trap component or tool that's not actually part of the panel code? If its the latter, it might be more appropriate to test the focus trap component if it is an internally developed one.

Reply:
This is meant to test tab order, but it wouldn't be able to effectively test focus trap anyway because there's no interactive elements outside of this panel in the mock/render. 🤔

Reply:
Do you expect the tab order to work any differently than the default browser behavior? If not, this test is testing jsdom instead of any panel logic. 

It looks like the `Portal` component relies on a package called `focus-trap-react` so my answer would not test for focus trapping. However, I see that the trap is activated by the `focusTrap` prop on the `Portal` component so that might be a reason to test even though it is just passing the prop directly to the focus trap. It's your call.

My general guidance is to only test logic that is located in the exact file that you have imported to test (`PanelV2`). If the logic is located in another file in your project, then that file should be tested separately (`Portal`). If the logic comes from a third-party package or library, then it is their responsibility to test their own code (`focus-trap-react`) not yours.

Reply:
Thanks that makes sense -- do you still think this test is worthwhile though, as-is? (And then something separate for focus trap, in the Portal tests most likely)

Reply:
I think it's good. I just realized that it tabs to the content area first, which makes sense why you'd want to tab through.